cachito/workers/pkg_managers/gomod.py

```diff
@@ -316,6 +311,82 @@ def resolve_gomod(app_source_path, request, dep_replacements=None, git_dir_path=
         return {"module": module, "module_deps": module_level_deps, "packages": packages}
 
 
+def _should_vendor_deps(flags: List[str], app_dir: str, strict: bool) -> Tuple[bool, bool]:
+    """
+    Determine if Cachito should vendor dependencies and if it is allowed to make changes.
+
+    This is based on the presence of flags:
+    - gomod-vendor-check => should vendor, can only make changes if vendor dir does not exist
+    - gomod-vendor => should vendor, can make changes
+
+    :param flags: flags from the Cachito request
+    :param app_dir: absolute path to the app directory
+    :param strict: fail the request if the vendor dir is present but the flags are not?
+    :return: (should vendor: bool, allowed to make changes: bool)
+    :raise ValidationError: if the vendor dir is present, the flags are not and we are strict
+    """
+    vendor = Path(app_dir) / "vendor"
+
+    if "gomod-vendor-check" in flags:
+        return True, not vendor.exists()
+    elif "gomod-vendor" in flags:
+        return True, True
+
+    if strict and vendor.is_dir():
```

Comment:
If `worker_config.cachito_gomod_strict_vendor` is False, Cachito will run `go mod download` eventually. What and how will it affect a repository having the `vendor` directory?

Reply:
I believe `go mod download` does not make changes in the repository. I could be wrong, but that's my understanding.